pkg/scheduler/framework/plugins/apienablement/api_enablement.go

```diff
@@ -55,6 +56,10 @@ func (p *APIEnablement) Filter(
 	cluster *clusterv1alpha1.Cluster,
 ) *framework.Result {
 	if !helper.IsAPIEnabled(cluster.Status.APIEnablements, bindingSpec.Resource.APIVersion, bindingSpec.Resource.Kind) {
+		// skip the filter if the cluster is already in the list of scheduling results and its API enablements are incomplete
+		if bindingSpec.TargetContains(cluster.Name) && !util.IsClusterCompleteAPIEnablements(&cluster.Status) {
+			return framework.NewResult(framework.Success)
+		}
```

Comment:
I've written a version like this, so we can see if it's easier to understand.
```go
if helper.IsAPIEnabled(cluster.Status.APIEnablements, bindingSpec.Resource.APIVersion, bindingSpec.Resource.Kind) {
		klog.V(2).Infof("Cluster(%s) is enable with the API(%s, kind=%s)",
			cluster.Name, bindingSpec.Resource.APIVersion, bindingSpec.Resource.Kind)
		return framework.NewResult(framework.Success)
	}
	klog.V(2).Infof("Cluster(%s) is not enable as missing API(%s, kind=%s)",
		cluster.Name, bindingSpec.Resource.APIVersion, bindingSpec.Resource.Kind)

	if util.IsClusterCompleteAPIEnablements(&cluster.Status) {
		klog.V(2).Infof("Cluster(%s) has completed APIEnablement", cluster.Name)
		return framework.NewResult(framework.Unschedulable, "cluster(s) did not have the API resource")
	}
	klog.V(2).Infof("Cluster(%s) has uncompleted APIEnablement", cluster.Name)

	if bindingSpec.TargetContains(cluster.Name) {
		klog.V(2).Infof("However the last scheduling result contains the cluster(%s), just pass it.", cluster.Name)
		return framework.NewResult(framework.Success)
	}
	klog.V(2).Infof("And the last scheduling result doesn't contains the cluster(%s).", cluster.Name)

	return framework.NewResult(framework.Unschedulable, "cluster(s) did not have the API resource")
```

Reply:
I like it, but just feel too much logs, how about this one:
```golang
	if helper.IsAPIEnabled(cluster.Status.APIEnablements, bindingSpec.Resource.APIVersion, bindingSpec.Resource.Kind) {
		return framework.NewResult(framework.Success)
	}

	// Let the cluster pass if it is already on the list of schedule result and the cluster's
	// API enablements is incomplete, to avoid the issue that cluster be accidentally removed
	// due to untrusted API enablements.
	if bindingSpec.TargetContains(cluster.Name) &&
		!meta.IsStatusConditionTrue(cluster.Status.Conditions, clusterv1alpha1.ClusterConditionCompleteAPIEnablements) {
		return framework.NewResult(framework.Success)
	}

	klog.V(2).Infof("Cluster(%s) not fit as missing API(%s, kind=%s)", cluster.Name, bindingSpec.Resource.APIVersion, bindingSpec.Resource.Kind)

	return framework.NewResult(framework.Unschedulable, "cluster(s) did not have the API resource")
```